README.md
# datWeather

history.txt
   51  git clean -f
   52  git status
   53  git clean
   54  git clean -inf
   55  git status
   56  git revert
   57  git rebase origin master
   58  git status
   59  git clean
   60  git reset
   61  git reset
   62  git status
   63  git stash
   64  git revert HEAD
   65  git fetch origin master
   66  git status
   67  git add .
   68  git stash
   69  git pull
   70  npm install
   71  cd test/
   72  ls
   73  test.js
   74  node test.js 
   75  npm i -g mocha
   76  npm i -g chai
   77  node test.js 
   78  git pull
   79  git status
   80  git add -A
   81  git status
   82  git stash
   83  clone git@github.com:the-Coding-Boot-Camp-at-UT/07-16-TR-Class-Content.git
   84  git clone git@github.com:the-Coding-Boot-Camp-at-UT/07-16-TR-Class-Content.git
   85  git clone https://github.com/the-Coding-Boot-Camp-at-UT/07-16-TR-Class-Content.git
   86  git clone git@github.com:construktivist/datWeather.git
   87  clip < ~/.ssh/id_rsa.pub
   88  ssh-keygen -t rsa -b 4096 -C "bradleynielsen@gmail.com"
   89  clip < ~/.ssh/id_rsa.pub
   90  git clone git@github.com:construktivist/datWeather.git
   91  cd datWeather/
   92  ls
   93  ls
   94  node app.js 
   95  git branch
   96  git status
   97  git status
   98  git add .
   99  git commit
  100  git push origin API
  101  git status
  102  eixt
  103  exit
  104  git status
  105  git pull
  106  git checkout origin api
  107  git status
  108  git add -A
  109  git commit -m "just before the sql is done"
  110  git push
  111  git add -A
  112  git commit -m "just before the sql is done"
  113  git push
  114  git status
  115  git pull
  116  git status
  117  git branch
  118  git push origin
  119  touch .gitignore
  120  git push origin API --Force
  121  git push origin API --force
  122  git checkout test
  123  git status
  124  git branch
  125  explorer .
  126  git stash
  127  git checkout origin test
  128  cd ..
  129  ls
  130  cd d
  131  vi desktop.ini 
  132  rm desktop.ini 
  133  ls
  134  cd datWeather/
  135  ls
  136  cd test/
  137  git clone git@github.com:construktivist/datWeather.git
  138  ls
  139  cd datWeather/
  140  ls
  141  cd db/
  142  ls
  143  cd datWeather/
  144  ls
  145  git status
  146  ls
  147  ls v
  148  ls -v
  149  ls l
  150  ls -l
  151  git status
  152  cd API/
  153  git rm cd ..
  154  ls
  155  ..
  156  cd ..
  157  ls
  158  git rm API/*
  159  git pull origin master
  160  git add -a
  161  git add -A
  162  git status
  163  git commit -m 'updating test'
  164  git pust origin test
  165  git push origin test
  166  touch index.sql
  167  git status
  168  git pull origin master
  169  git pull
  170  git pull origin master
  171  git status
  172  git push origin test
  173  git status
  174  git branch
  175  git add .
  176  git commit "starting index for sql"
  177  git commit -m "starting index for sql"
  178  git push
  179  git branch
  180  git checkout sqlindex
  181  git branch sqlindex
  182  git branch
  183  git checkout sqlindex 
  184  git  status
  185  git add .
  186  git commit -m "added index workd; adding PK to primary DB"
  187  git push 
  188  git push origin sqlindex
  189  git checkout test
  190  git merge origin master
  191  git push origin master
  192  git checkout sqlindex 
  193  git fetch origin master
  194  git status
  195  git add .
  196  git status
  197  git pull test
  198  git pull origin test
  199  git staus
  200  y
  201  git status
  202  clear
  203  git add .
  204  git commit -m "consolidating file, prep for breakage"
  205  git add -A
  206  git commit -m "consolidating file, prep for breakage"
  207  git add -A
  208  git commit -m "consolidating file, prep for breakage"
  209  git add -A
  210  git commit -m "consolidating file workd"
  211  git push origin sqlindex 
  212  git status
  213  git status
  214  git status
  215  git status
  216  git pull origin test
  217  git checkout test
  218  git branch sqlstats
  219  git checkout sqlstats 
  220  node server.js
  221  node /bin/www server.js
  222  node /bin/www
  223  node /bin www
  224  npm install
  225  node /bin www
  226  cd bin
  227  ls
  228  node www
  229  node www
  230  node www
  231  node www
  232  ls
  233  cd ..
  234  ls
  235  npm install mysql --save
  236  npm install sequelize --save
  237  node www
  238  cd config/
  239  ls
  240  cd ..
  241  ls
  242  cd bin/
  243  ls
  244  node www
  245  fs
  246  node fs
  247  npm fs
  248  npm install fs -g
  249  node www
  250  node www
  251  node www
  252  node www
  253  node www
  254  node www
  255  node www
  256  node www
  257  node www
  258  node www
  259  node server.js 
  260  npm install
  261  server.js
  262  node server.js 
  263  ls
  264  cd ..
  265  cd ..
  266  ls
  267  cd ..
  268  cd ..
  269  sl
  270  ls
  271  cd ..
  272  cd ..
  273  ls
  274  cd ..
  275  cd ..
  276  ls
  277  cd datWeather/
  278  ls
  279  cd dev/
  280  ls
  281  cd datWeather/
  282  ls
  283  git status
  284  cd bin/
  285  ls
  286  node www
  287  git status
  288  git add .
  289  git add . 
  290  git commit -m "the route is working"
  291  git add . 
  292  git commit -m " dates and temps sucka"
  293  git add . 
  294  git commit -m " working on adding the calculations file next"
  295  git push
  296  git push origin sqlstats 

routes/index.js
var express = require('express');
var router = express.Router();
var models = require('../models');
var stats = require("stats-lite");

// var sequelizeConnection = models.sequelize;
// sequelizeConnection.sync();



/* GET home page. */
router.get('/', function(req, res) {
  res.render('index', {
  	City: 'New York' ,
  	springTemperature: "60 - 82",
  	springPercipitation: "Wet",
  	springHumidity: "10% - 20%",
  	springWind: "12 mph",


  	summerTemperature: "86 - 103",
  	summerPercipitation: "Very Dry",
  	summerHumidity: "0% - 3%",
  	summerWind: "14 mph",

  	fallTemperature: "60 - 78",
  	fallPercipitation: "Very Wet",
  	fallHumidity: "1% - 2%",
  	fallWind: "18 mph",

  	winterTemperature: "13 - 67",
  	winterPercipitation: "Dry",
  	winterHumidity: "0% - 2%",
  	winterWind: "8 mph"

    });
});

var springStats = {};
var summerStats = {};
var fallStats = {};
var winterStats = {};

//================
//Get all the data
//================
router.post("/city", function(request, response) {

    var cityName = request.body.city;
    console.log(cityName);
    var condition = cityName.toUpperCase();

    console.log("condition:", condition);


    //Spring
    models.spring.findAll({
        attributes: ['HOURLYDRYBULBTEMPF', 'HOURLYPrecip', 'HOURLYWindSpeed']
    }).then(function (data) {
        // console.log(response.json(data));
        // return response.json(data);


        var springTempArray = [];
        var springPrecipArray = [];
        var springWindArray = [];

        //Loop over the sql array
        for (var i = 0; i < data.length; i++) {
            springTempArray.push(data[i].HOURLYDRYBULBTEMPF);
            springPrecipArray.push(data[i].HOURLYPrecip);
            springWindArray.push(data[i].HOURLYWindSpeed);
        }

        //
        //Temps
        //

        console.log("mean: %s", stats.mean(springTempArray));
        console.log("median: %s", stats.median(springTempArray));
        console.log("mode: %s", stats.mode(springTempArray));
        console.log("standard deviation: %s", stats.stdev(springTempArray));
        springStats.springTempMean = stats.mean(springTempArray);
        springStats.springTempMedian = stats.median(springTempArray);
        springStats.springTempMode = stats.mode(springTempArray);
        springStats.springTempStdev = stats.stdev(springTempArray);

        //
        //Rains
        //

        console.log("mean: %s", stats.mean(springPrecipArray));
        console.log("median: %s", stats.median(springPrecipArray));
        console.log("mode: %s", stats.mode(springPrecipArray));
        console.log("standard deviation: %s", stats.stdev(springPrecipArray));
        springStats.springRainMean = stats.mean(springPrecipArray);
        springStats.springRainMedian = stats.median(springPrecipArray);
        springStats.springRainMode = stats.mode(springPrecipArray);
        springStats.springRainStdev = stats.stdev(springPrecipArray);


        //
        //Winds
        //

        console.log("mean: %s", stats.mean(springWindArray));
        console.log("median: %s", stats.median(springWindArray));
        console.log("mode: %s", stats.mode(springWindArray));
        console.log("standard deviation: %s", stats.stdev(springWindArray));
        springStats.springWindMean = stats.mean(springWindArray);
        springStats.springWindMedian = stats.median(springWindArray);
        springStats.springWindMode = stats.mode(springWindArray);
        springStats.springWindStdev = stats.stdev(springWindArray);


    }).then(function (data) {
        //Summer
        models.summer.findAll({
            attributes: ['HOURLYDRYBULBTEMPF', 'HOURLYPrecip', 'HOURLYWindSpeed']
        }).then(function (data) {
            // console.log(response.json(data));
            // return response.json(data);

            var summerTempArray = [];
            var summerPrecipArray = [];
            var summerWindArray = [];


            //Loop over the sql array
            for (var i = 0; i < data.length; i++) {
                summertempArray.push(data[i].HOURLYDRYBULBTEMPF);
                summerprecipArray.push(data[i].HOURLYPrecip);
                summerwindArray.push(data[i].HOURLYWindSpeed);
            }

            //
            //Temps
            //

            console.log("mean: %s", stats.mean(summerTempArray));
            console.log("median: %s", stats.median(summerTempArray));
            console.log("mode: %s", stats.mode(summerTempArray));
            console.log("standard deviation: %s", stats.stdev(summerTempArray));
            springStats.springTempMean = stats.mean(summerTempArray);
            springStats.springTempMedian = stats.median(summerTempArray);
            springStats.springTempMode = stats.mode(summerTempArray);
            springStats.springTempStdev = stats.stdev(summerTempArray);

            //
            //Rains
            //

            console.log("mean: %s", stats.mean(summerPrecipArray));
            console.log("median: %s", stats.median(summerPrecipArray));
            console.log("mode: %s", stats.mode(summerPrecipArray));
            console.log("standard deviation: %s", stats.stdev(summerPrecipArray));
            springStats.springRainMean = stats.mean(summerPrecipArray);
            springStats.springRainMedian = stats.median(summerPrecipArray);
            springStats.springRainMode = stats.mode(summerPrecipArray);
            springStats.springRainStdev = stats.stdev(summerPrecipArray);


            //
            //Winds
            //

            console.log("mean: %s", stats.mean(summerWindArray));
            console.log("median: %s", stats.median(summerWindArray));
            console.log("mode: %s", stats.mode(summerWindArray));
            console.log("standard deviation: %s", stats.stdev(summerWindArray));
            springStats.springWindMean = stats.mean(summerWindArray);
            springStats.springWindMedian = stats.median(summerWindArray);
            springStats.springWindMode = stats.mode(summerWindArray);
            springStats.springWindStdev = stats.stdev(summerWindArray);
        })
    }).then(function (data) {
        //Fall
        models.fall.findAll({
            attributes: ['HOURLYDRYBULBTEMPF', 'HOURLYPrecip', 'HOURLYWindSpeed']
        }).then(function (data) {
            // console.log(response.json(data));
            // return response.json(data);

            var tempArray = [];
            var precipArray = [];
            var windArray = [];


            //Loop over the sql array
            for (var i = 0; i < data.length; i++) {
                tempArray.push(data[i].HOURLYDRYBULBTEMPF);
                precipArray.push(data[i].HOURLYPrecip);
                windArray.push(data[i].HOURLYWindSpeed);
            }

            //
            //Temps
            //
            fallStats.fallTempMean = stats.mean(fallTempArray);
            fallStats.fallTempMedian = stats.median(fallTempArray);
            fallStats.fallTempMode = stats.mode(fallTempArray);
            fallStats.fallTempStdev = stats.stdev(fallTempArray);

            //
            //Rains
            //
            fallStats.fallPercipMean = stats.mean(fallPercipArray);
            fallStats.fallPercipMedian = stats.median(fallPercipArray);
            fallStats.fallPercipMode = stats.mode(fallPercipArray);
            fallStats.fallPercipStdev = stats.stdev(fallPercipArray);

            //
            //Winds
            //
            fallStats.fallWindMean = stats.mean(fallWindArray);
            fallStats.fallWindMedian = stats.median(fallWindArray);
            fallStats.fallWindMode = stats.mode(fallWindArray);
            fallStats.fallWindStdev = stats.stdev(fallWindArray);
        })
    }).then(function (data) {
        //    Winter
        models.winter.findAll({
            attributes: ['HOURLYDRYBULBTEMPF', 'HOURLYPrecip', 'HOURLYWindSpeed']
        }).then(function (data) {
            // console.log(response.json(data));
            // return response.json(data);

            var tempArray = [];
            var precipArray = [];
            var windArray = [];


            //Loop over the sql array
            for (var i = 0; i < data.length; i++) {
                tempArray.push(data[i].HOURLYDRYBULBTEMPF);
                precipArray.push(data[i].HOURLYPrecip);
                windArray.push(data[i].HOURLYWindSpeed);
            }

            //
            //Temps
            //
            winterStats.winterTempMean = stats.mean(winterTempArray);
            winterStats.winterTempMedian = stats.median(winterTempArray);
            winterStats.winterTempMode = stats.mode(winterTempArray);
            winterStats.winterTempStdev = stats.stdev(winterTempArray);

            //
            //Rains
            //
            winterStats.winterPercipMean = stats.mean(winterPercipArray);
            winterStats.winterPercipMedian = stats.median(winterPercipArray);
            winterStats.winterPercipMode = stats.mode(winterPercipArray);
            winterStats.winterPercipStdev = stats.stdev(winterPercipArray);

            //
            //Winds
            //
            winterStats.winterWindMean = stats.mean(winterWindArray);
            winterStats.winterWindMedian = stats.median(winterWindArray);
            winterStats.winterWindMode = stats.mode(winterWindArray);
            winterStats.winterWindStdev = stats.stdev(winterWindArray);
        })
    }).then(function (springStats, summerStats, fallStats, winterStats) {
        response.render("index", {
            City: condition,
            springTemperature: "60 - 82",
            springPercipitation: "Wet",
            springHumidity: "10% - 20%",
            springWind: "12 mph",


            summerTemperature: "86 - 103",
            summerPercipitation: "Very Dry",
            summerHumidity: "0% - 3%",
            summerWind: "14 mph",

            fallTemperature: "60 - 78",
            fallPercipitation: "Very Wet",
            fallHumidity: "1% - 2%",
            fallWind: "18 mph",

            winterTemperature: "13 - 67",
            winterPercipitation: "Dry",
            winterHumidity: "0% - 2%",
            winterWind: "8 mph"

        });
    })




});    //END OF ROUTE





//
//Spring Route
//
router.get('/austin/spring', function(request, response) {
    models.spring.findAll({
        attributes: ['HOURLYDRYBULBTEMPF', 'HOURLYPrecip','HOURLYWindSpeed' ]
    }).then(function(data) {
        // console.log(response.json(data));
        // return response.json(data);

        var tempArray = [];
        var precipArray = [];
        var windArray = [];


        //Loop over the sql array
        for (var i = 0; i < data.length; i++) {
            tempArray.push(data[i].HOURLYDRYBULBTEMPF);
            precipArray.push(data[i].HOURLYPrecip);
            windArray.push(data[i].HOURLYWindSpeed);
        }

        //
        //Temps
        //
        console.log("=======");
        console.log("Temps");
        console.log("=======");

        console.log("mean: %s", stats.mean(tempArray));
        console.log("median: %s", stats.median(tempArray));
        console.log("mode: %s", stats.mode(tempArray));
        console.log("standard deviation: %s", stats.stdev(tempArray));

        //
        //Rains
        //
        console.log("=======");
        console.log("Rains");
        console.log("=======");
        console.log("mean: %s", stats.mean(precipArray));
        console.log("median: %s", stats.median(precipArray));
        console.log("mode: %s", stats.mode(precipArray));
        console.log("standard deviation: %s", stats.stdev(precipArray));

        //
        //Winds
        //
        console.log("=======");
        console.log("Winds");
        console.log("=======");
        console.log("mean: %s", stats.mean(windArray));
        console.log("median: %s", stats.median(windArray));
        console.log("mode: %s", stats.mode(windArray));
        console.log("standard deviation: %s", stats.stdev(windArray));
    })
});


//
//summer Route
//
router.get('/austin/summer', function(request, response) {
    models.summer.findAll({
        attributes: ['HOURLYDRYBULBTEMPF', 'HOURLYPrecip','HOURLYWindSpeed' ]
    }).then(function(data) {
        // console.log(response.json(data));
        // return response.json(data);

        var tempArray = [];
        var precipArray = [];
        var windArray = [];


        //Loop over the sql array
        for (var i = 0; i < data.length; i++) {
            tempArray.push(data[i].HOURLYDRYBULBTEMPF);
            precipArray.push(data[i].HOURLYPrecip);
            windArray.push(data[i].HOURLYWindSpeed);
        }

        //
        //Temps
        //
        console.log("=======");
        console.log("Temps");
        console.log("=======");

        console.log("mean: %s", stats.mean(tempArray));
        console.log("median: %s", stats.median(tempArray));
        console.log("mode: %s", stats.mode(tempArray));
        console.log("standard deviation: %s", stats.stdev(tempArray));

        //
        //Rains
        //
        console.log("=======");
        console.log("Rains");
        console.log("=======");
        console.log("mean: %s", stats.mean(precipArray));
        console.log("median: %s", stats.median(precipArray));
        console.log("mode: %s", stats.mode(precipArray));
        console.log("standard deviation: %s", stats.stdev(precipArray));

        //
        //Winds
        //
        console.log("=======");
        console.log("Winds");
        console.log("=======");
        console.log("mean: %s", stats.mean(windArray));
        console.log("median: %s", stats.median(windArray));
        console.log("mode: %s", stats.mode(windArray));
        console.log("standard deviation: %s", stats.stdev(windArray));
    })
});

router.post("/city", function(request, response){
	var cityName = request.body.city;
	console.log(cityName);
	var condition = cityName.toUpperCase();

	console.log("condition:", condition);

	response.render("index", {
		City: condition,
	})
})







router.get('/:austin', function(request, response) {
console.log("help");
    models.Campmabrymin.findAll({
        attributes: ['DATE', 'HOURLYDRYBULBTEMPF']
    }).then(function(data){
        // var datajson = json.parse(data);
        // response.render('index', {
        //     data: datajson
        // });
        var dataJson = json.parse(data);
        response.render('index', {
            data: dataJson
        });

	});
});






//
//fall Route
//
router.get('/austin/fall', function(request, response) {
    models.fall.findAll({
        attributes: ['HOURLYDRYBULBTEMPF', 'HOURLYPrecip','HOURLYWindSpeed' ]
    }).then(function(data) {
        // console.log(response.json(data));
        // return response.json(data);

        var tempArray = [];
        var precipArray = [];
        var windArray = [];


        //Loop over the sql array
        for (var i = 0; i < data.length; i++) {
            tempArray.push(data[i].HOURLYDRYBULBTEMPF);
            precipArray.push(data[i].HOURLYPrecip);
            windArray.push(data[i].HOURLYWindSpeed);
        }

        //
        //Temps
        //
        console.log("=======");
        console.log("Temps");
        console.log("=======");

        console.log("mean: %s", stats.mean(tempArray));
        console.log("median: %s", stats.median(tempArray));
        console.log("mode: %s", stats.mode(tempArray));
        console.log("standard deviation: %s", stats.stdev(tempArray));

        //
        //Rains
        //
        console.log("=======");
        console.log("Rains");
        console.log("=======");
        console.log("mean: %s", stats.mean(precipArray));
        console.log("median: %s", stats.median(precipArray));
        console.log("mode: %s", stats.mode(precipArray));
        console.log("standard deviation: %s", stats.stdev(precipArray));

        //
        //Winds
        //
        console.log("=======");
        console.log("Winds");
        console.log("=======");
        console.log("mean: %s", stats.mean(windArray));
        console.log("median: %s", stats.median(windArray));
        console.log("mode: %s", stats.mode(windArray));
        console.log("standard deviation: %s", stats.stdev(windArray));
    })
});


//
//winter Route
//
router.get('/austin/winter', function(request, response) {
    models.winter.findAll({
        attributes: ['HOURLYDRYBULBTEMPF', 'HOURLYPrecip','HOURLYWindSpeed' ]
    }).then(function(data) {
        // console.log(response.json(data));
        // return response.json(data);

        var tempArray = [];
        var precipArray = [];
        var windArray = [];


        //Loop over the sql array
        for (var i = 0; i < data.length; i++) {
            tempArray.push(data[i].HOURLYDRYBULBTEMPF);
            precipArray.push(data[i].HOURLYPrecip);
            windArray.push(data[i].HOURLYWindSpeed);
        }

        //
        //Temps
        //
        console.log("=======");
        console.log("Temps");
        console.log("=======");

        console.log("mean: %s", stats.mean(tempArray));
        console.log("median: %s", stats.median(tempArray));
        console.log("mode: %s", stats.mode(tempArray));
        console.log("standard deviation: %s", stats.stdev(tempArray));

        //
        //Rains
        //
        console.log("=======");
        console.log("Rains");
        console.log("=======");
        console.log("mean: %s", stats.mean(precipArray));
        console.log("median: %s", stats.median(precipArray));
        console.log("mode: %s", stats.mode(precipArray));
        console.log("standard deviation: %s", stats.stdev(precipArray));

        //
        //Winds
        //
        console.log("=======");
        console.log("Winds");
        console.log("=======");
        console.log("mean: %s", stats.mean(windArray));
        console.log("median: %s", stats.median(windArray));
        console.log("mode: %s", stats.mode(windArray));
        console.log("standard deviation: %s", stats.stdev(windArray));
    })
});


//
//Spring Route
//
router.get('/austin/spring', function(request, response) {
    models.spring.findAll({
        attributes: ['HOURLYDRYBULBTEMPF', 'HOURLYPrecip','HOURLYWindSpeed' ]
    }).then(function(data) {
        // console.log(response.json(data));
        // return response.json(data);

        var tempArray = [];
        var precipArray = [];
        var windArray = [];


        //Loop over the sql array
        for (var i = 0; i < data.length; i++) {
            tempArray.push(data[i].HOURLYDRYBULBTEMPF);
            precipArray.push(data[i].HOURLYPrecip);
            windArray.push(data[i].HOURLYWindSpeed);
        }

        //
        //Temps
        //
        console.log("=======");
        console.log("Temps");
        console.log("=======");

        console.log("mean: %s", stats.mean(tempArray));
        console.log("median: %s", stats.median(tempArray));
        console.log("mode: %s", stats.mode(tempArray));
        console.log("standard deviation: %s", stats.stdev(tempArray));

        //
        //Rains
        //
        console.log("=======");
        console.log("Rains");
        console.log("=======");
        console.log("mean: %s", stats.mean(precipArray));
        console.log("median: %s", stats.median(precipArray));
        console.log("mode: %s", stats.mode(precipArray));
        console.log("standard deviation: %s", stats.stdev(precipArray));

        //
        //Winds
        //
        console.log("=======");
        console.log("Winds");
        console.log("=======");
        console.log("mean: %s", stats.mean(windArray));
        console.log("median: %s", stats.median(windArray));
        console.log("mode: %s", stats.mode(windArray));
        console.log("standard deviation: %s", stats.stdev(windArray));
    })
});


module.exports = router;